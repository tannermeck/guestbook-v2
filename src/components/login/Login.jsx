import style from './login.css';

export default function Login(){
    return (
        <fieldset>
            <legend>Sign In</legend>
            <form>
                <label htmlFor="username">
                Username
                </label>
                <input className={style.inputBox}
                id="username"
                type="text"
                name="username"
                required
                />
                <label htmlFor="password">
                Password
                </label>
                <input
                id="password"
                type="password"
                name="password"
                required
                />
                <button
                type="submit"
                >
                Sign In
                </button>
            </form>
            <p>Error message goes here</p>
        </fieldset>
    )
}