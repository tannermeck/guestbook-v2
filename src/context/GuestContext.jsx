import { useContext, useState, createContext } from 'react';

const GuestContext = createContext();

const GuestProvider = ({children}) => {
    const [guest, setGuest] = useState('')

    return (
        <GuestContext.Provider value={{guest, setGuest}}>
            {children}
        </GuestContext.Provider>
    )
}
const useGuest = () => {
    const context = useContext(GuestContext);
    if (context === undefined){
        throw new Error('useGuest hook must be called inside of a GuessContext Provider');
    }
    return context;
}
export { GuestProvider, useGuest}