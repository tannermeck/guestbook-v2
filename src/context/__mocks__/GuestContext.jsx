import { createContext, useState } from "react";

const GuestContext = createContext();

const GuestProvider = ({mockUser, children}) => {
    const [guest, setGuest] = useState(
    mockUser ? 'tanner' : '' )

    return (
        <GuestContext.Provider value={{guest, setGuest}}>
            {children}
        </GuestContext.Provider>
    )
}
export { GuestProvider, GuestContext }