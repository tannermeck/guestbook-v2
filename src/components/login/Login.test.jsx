import { render } from '@testing-library/react';
import { GuestProvider } from '../../context/GuestContext';
import Login from './Login';
import { MemoryRouter, Switch, Route } from 'react-router-dom';

it('should render a login view', () => {
    const { container } = render(
    <GuestProvider>
        <MemoryRouter initialEntries={["/login"]}>
            <Switch>
                <Route exact path='/login'>
                    <Login />
                </Route>
            </Switch>
        </MemoryRouter>
    </GuestProvider>)
    expect(container).toMatchSnapshot();
    
})