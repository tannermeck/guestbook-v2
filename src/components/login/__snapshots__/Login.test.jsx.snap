// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a login view 1`] = `
<div>
  <fieldset>
    <legend>
      Sign In
    </legend>
    <form>
      <label
        for="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
    <p />
  </fieldset>
</div>
`;
